src/Dialog/Dialog.tsx

```diff
@@ -302,6 +302,7 @@ const _Dialog = React.forwardRef<HTMLDivElement, React.PropsWithChildren<DialogP
             role={role}
             aria-labelledby={dialogLabelId}
             aria-describedby={dialogDescriptionId}
+            aria-modal
```

Comment:
I considered adding a prop to allow users to override this, but since we don't allow them to override the focus zone or exit behavior, there is no valid scenario in which this dialog is not modal.